### PR TITLE
feature(main): Add tutorial section

### PR DESCRIPTION
**Pull request by kjdelisle**
connected to https://github.com/strongloop/loopback-next/issues/670

This is a tutorial on how to reproduce this repository's application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,3 +32,383 @@ cd loopback4-example-getting-started && npm install
...
+  constructor(
+    @repository(TodoRepository.name) protected todoRepo: TodoRepository,
+  ) {}
+  @post('/todo')
+  @param.body('todo', TodoSchema)
+  async createTodo(todo: Todo) {
+    if (!todo.title) {
+      return Promise.reject(new HttpErrors.BadRequest('title is required'));
+    }
+    const result = await this.todoRepo.create(todo);
```
Comment (b-admike) [inline README.md]: can we just do `return await this.todoRepo.create(todo);` here?

**Review by b-admike — APPROVED** [review]
Awesome tutorial! I want to try it out, but otherwise LGTM.

File: README.md (hunk 1 above)
Comment (kjdelisle) [reply README.md]: Yep! That actually needs fixing in the real example code as well, though that'll be a different PR.

File: README.md [hunk 2]
```diff
@@ -32,3 +32,382 @@ cd loopback4-example-getting-started && npm install
...
+not to enable certain project features (loopback's build, tslint, mocha, etc.),
+leave them all enabled.
+
+<!-- TODO: Add screenshot of terminal here to illustrate what we mean. -->
+
+### Adding Legacy Juggler Capabilities
+Jump into the directory for your new application. You'll see a folder structure
+similar to this:
+```
+dist\
```
Comment (jannyHou) [inline README.md:57]: nitpick... any reason the folder signature uses `\` not `/`

File: README.md [hunk 3]
```diff
@@ -32,3 +32,382 @@ cd loopback4-example-getting-started && npm install
...
+  properties: {
+    id: {
+      type: 'number',
+      description: 'ID number of the Todo entry.'
+    },
+    title: {
+      type: 'string',
+      description: 'Title of the Todo entry.'
+    },
+    desc: {
```
Comment (jannyHou) [inline README.md:206]: copy typo? It should be something like:

> {
 type: 'string',
 description: 'a description that details what the todo is all about'
}

File: README.md [hunk 4]
```diff
@@ -32,3 +32,382 @@ cd loopback4-example-getting-started && npm install
...
+      type: 'number',
+      description: 'ID number of the Todo entry.'
+    },
+    isComplete: {
+      type: 'boolean',
+      description: 'Whether or not the Todo entry is complete.'
+    }
+  },
+  required: ['title'],
```
Comment (jannyHou) [inline README.md:215]: Is the `id` required by default or we need to specify it here as well?

File: README.md [hunk 5]
```diff
@@ -32,3 +32,382 @@ cd loopback4-example-getting-started && npm install
...
+```ts
+import { DefaultCrudRepository, DataSourceType } from '@loopback/repository';
+import { Todo } from '../models';
+import { inject } from '@loopback/core';
+
+export class TodoRepository extends DefaultCrudRepository<
+  Todo,
+  typeof Todo.prototype.id
+> {
+  constructor(@inject('datasource') protected datasource: DataSourceType) {
```
Comment (jannyHou) [inline README.md:275]: nitpick, maybe refer to [application.ts](https://github.com/strongloop/loopback4-example-getting-started/pull/11/files#diff-04c6e90faac2675aa89e2176d2eec7d8R395) so readers know where is the `datasource` bound to. Just feel a little bit weird to see the injection code first...or maybe that's the way how dependency injection works?

**Review by jannyHou — COMMENTED** [review]
Wait...still reviewing :(

**Pull request merged**